stock/redis_store.go

```diff
@@ -13,6 +13,24 @@ import (
 	"github.com/valyala/fasthttp"
 )
 
+// Script with condition checking
+var decrByXX = redis.NewScript(`
+	local value = redis.call("GET", KEYS[1])
+	local stockValue
+	local priceValue
+	for iter in string.gmatch(value, "(%d+)}") do
+    	stockValue=iter
+	end
+	for iter in string.gmatch(value, "(%d+),") do
+    	priceValue=iter
+	end
+	local updatedValue = tonumber(stockValue) - ARGV[1]
+	if  updatedValue > -1 then
+  		return redis.call("SET", KEYS[1], string.format('{%s: %d,%s: %d}','"price"',tonumber(priceValue),'"stock"',updatedValue))
```

Comment:
I'd put the `%s` formatted parts directly in the format string.
```suggestion
  		return redis.call("SET", KEYS[1], string.format('{"price": %d, "stock": %d}', tonumber(priceValue), updatedValue))
```

Reply:
Incorporated these changes